tools/ports/__init__.py

```diff
@@ -135,9 +135,10 @@ def build_port(src_dir, output_path, port_name, includes=[], flags=[], cxxflags=
       srcs = [os.path.join(src_dir, s) for s in srcs]
     else:
       srcs = []
-      for root, _, files in os.walk(src_dir, topdown=False):
-        if any((excluded in root) for excluded in exclude_dirs):
-          continue
+      for root, dirs, files in os.walk(src_dir):
+        for ex in exclude_dirs:
+          if ex in dirs:
+            dirs.remove(ex)
```

Comment:
Interesting, TIL that you can modify `dirs` in this way and it affects the walk. Python is fancy...